src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs

```diff
@@ -52,11 +52,25 @@ public override TestResult Execute(TestExecutionContext context)
             if (Test.Fixture == null)
                 Test.Fixture = context.TestObject;
 
-            try
+            RunTestMethodInThreadAbortSafeZone(context, () =>
             {
                 BeforeTest(context);
-
                 context.CurrentResult = innerCommand.Execute(context);
+            });
+
+            if (context.ExecutionStatus != TestExecutionStatus.AbortRequested)
+            {
+                RunTestMethodInThreadAbortSafeZone(context, () => { AfterTest(context); });
+            }
+
+            return context.CurrentResult;
+        }
+
+        private static void RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)
```

Comment:
It took me a a few minutes to figure out what "thread-abort-safe zone" means, but it looks like it means that it's not inside a finally block. It could be called inside a finally block higher up the call stack though, in which case it would not be a thread-abort-safe zone despite the name. Am I understanding this properly?